lib/timers.js

```diff
@@ -229,7 +234,7 @@ function tryOnTimeout(timer, list) {
   timer._called = true;
   var threw = true;
   try {
-    timer._onTimeout();
+    ontimeout(timer);
```

Comment:
> but adding a `try-*` block certainly does cause a deopt for the containing function.

Right, which is why it was constrained to within one separate function.

I'm curious if a throw fork a user callback de-opts `ontimeout()` along the way though, since that could be somewhat unpleasant.

Maybe we shouldn't worry about it? Idk the sate of benchmarking this stuff is already so bad 😐